ocp-scripts/import-image-from-dockerhub.sh

```diff
@@ -0,0 +1,144 @@
+    echo -e "\033[92mDONE\033[39m: $1"
+}
+
+echo_warn(){
+    echo -e "\033[93mWARN\033[39m: $1"
+}
+
+echo_error(){
+    echo -e "\033[31mERROR\033[39m: $1"
+}
+
+echo_info(){
+    echo -e "\033[94mINFO\033[39m: $1"
+}
+
+ODS_AT_DOCKERHUB=docker.io/opendevstackorg
+IMAGE_NAME=
+IMAGE_VERSION=
+NAMESPACE=ods
+TARGET_IMAGE_STREAM=
+
+function usage {
+    printf "Import dockerhub image/tags.\n\n"
+    printf "This script will ask interactively for parameters by default.\n"
+    printf "However, you can also pass them directly. Usage:\n\n"
+    printf "\t-h|--help\t\tPrint usage\n"
+    printf "\t-v|--verbose\t\tEnable verbose mode\n"
+    printf "\t-on|--namespace\t\tNamespace, e.g. 'ods'\n"
+    printf "\t-in|--imagename\t\tImagename, e.g. 'ods-provisioning-app'\n"
+    printf "\t-it|--targetimagestream\t\tTargetImageName, e.g. 'ods-provisioning-app'\n"
+    printf "\t-iv|--imageversion\t\tImageversion, e.g. '2.x'\n"
+}
+
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+
+    -v|--verbose) set -x;;
+
+    -h|--help) usage; exit 0;;
+
+    -on|--namespace) NAMESPACE="$2"; shift;;
+    -on=*|--namespace=*) NAMESPACE="${1#*=}";;
+
+    -in|--imagename) IMAGE_NAME="$2"; shift;;
+    -in=*|--imagename=*) IMAGE_NAME="${1#*=}";;
+
+    -iv|--imageversion) IMAGE_VERSION="$2"; shift;;
+    -iv=*|--imageversion=*) IMAGE_VERSION="${1#*=}";;
+
+    -it|--targetimagestream) TARGET_IMAGE_STREAM="$2"; shift;;
+    -it=*|--targetimagestream=*) TARGET_IMAGE_STREAM="${1#*=}";;
+
+    *) echo_error "Unknown parameter passed: $1"; exit 1;;
+esac; shift; done
+
+if ! which jq >/dev/null; then
+    echo_warn "Binary 'jq' (https://stedolan.github.io/jq/) is not in your PATH. This will make the script less comfortable to use."
+    read -r -e -p "Continue anyway? [y/n] " input
+    if [ "${input:-""}" != "y" ]; then
+        exit 1
+    fi
+fi
+
+if [ -z "${NAMESPACE}" ]; then
+    odsprojectname="odsst-dev"
+    if [ -f "${ODS_CORE_DIR}/../ods-configuration/ods-core.env" ]; then
+        echo_info "Configuration located"
+        odsprojectname=$(grep NAMESPACE "${ODS_CORE_DIR}/../ods-configuration/ods-core.env" | cut -d "=" -f 2-)
+    fi
+    read -r -e -p "Enter ods central namespace [${odsprojectname}]: " input
+    if [[ -z "${input}" || $(echo -n $input | wc -m) == 0 ]]; then
+        NAMESPACE=${odsprojectname}
+    else
+        NAMESPACE=${input:-""}
+    fi
+fi
+
+if ! oc project ${NAMESPACE} -q; then
+    echo_error "OCP Project ${NAMESPACE} does NOT exist"
+    exit 1
+fi
+
+if [ -z "${IMAGE_NAME}" ]; then
+    read -r -e -p "Enter image name: " input
+    if [ -z "${input}" ]; then
+        echo_error "Image name is mandatory, aborting without it"
+        exit 1
+    else
+        IMAGE_NAME=${input:-""}
+    fi
+fi
+
+if [ -z "${TARGET_IMAGE_STREAM}" ]; then
+    read -r -e -p "Enter target image stream name: " input
+    if [ -z "${input}" ]; then
+        TARGET_IMAGE_STREAM=$IMAGE_NAME
+        echo "Target not set, defaulting to source: $IMAGE_NAME"

```

Comment:
I think it would be nice to display the default in the prompt, such as: `Enter target image stream name [foobar]: `. When the user hits enter, `foobar` is assigned.

Reply:
Ah -  I see you already used this pattern in line 121 :+1: